pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -409,11 +421,35 @@ private File downloadSegmentFromDeepStore(String segmentName, SegmentZKMetadata
       throws Exception {
     File tempRootDir = getTmpSegmentDataDir("tmp-" + segmentName + "-" + UUID.randomUUID());
     FileUtils.forceMkdir(tempRootDir);
+    if (_isSegmentDownloadUntarStreamed && zkMetadata.getCrypterName() == null) {
```

Comment:
so seems possible for me to configure streaming and encryption which is not a valid config but we do not log and just default to non streaming, should we log that in the else below?